app/services/remove_user_service.rb

```diff
@@ -0,0 +1,19 @@
+class RemoveUserService
+  include ServicePattern
+
+  def initialize(user, organisation)
+    @user = user
+    @organisation = organisation
+  end
+
+  attr_reader :user, :organisation
+
+  def call
+    membership = user.memberships.find_by(organisation:)
+    ActiveRecord::Base.transaction do
+      membership.destroy!
+      UserMailer.removal_email(user, organisation).deliver_later
+    end
```

Comment:
I suspect the database transaction has no effect here – since `membership.destroy!` is the only thing which hits the database. And once you've sent an email, you can't roll it back even as part of a transaction.

So I think it's safe to remove. But it causes no harm being here either.